diff --git a/config_space/sid/files/root/provision_extra.sh/AZURE b/config_space/sid/files/root/provision_extra.sh/AZURE
new file mode 100644
index 0000000..da60e12
--- /dev/null
+++ b/config_space/sid/files/root/provision_extra.sh/AZURE
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+if (( `systemctl list-units --state=running | grep -c uuidd` == 0 )); then
+	sudo apt install -y uuid-runtime
+	sudo systemctl start uuidd
+	sudo systemctl enable uuidd
+	sudo systemctl status uuidd
+fi
+
+sudo apt install -y bash-completion man-db manpages nftables rsync curl vim
diff --git a/config_space/sid/files/root/provision_locale_tz.sh/AZURE b/config_space/sid/files/root/provision_locale_tz.sh/AZURE
new file mode 100644
index 0000000..815bd07
--- /dev/null
+++ b/config_space/sid/files/root/provision_locale_tz.sh/AZURE
@@ -0,0 +1,27 @@
+#!/bin/bash
+
+uptime
+w
+last
+
+_tz='America/Los_Angeles'
+if [[ `cat /etc/timezone` != $_tz  && -f /usr/share/zoneinfo/$_tz ]]; then
+	sudo unlink /etc/localtime
+	echo $_tz | sudo tee /etc/timezone
+	sudo dpkg-reconfigure -f noninteractive tzdata
+fi
+
+_lang='en_US.UTF-8'
+if [[ "$LANG" != $_lang && `cat /etc/locale.gen` != $_lang ]]; then
+	grep $_lang /etc/locale.gen | sed 's/^# *//' | sudo tee /etc/locale.gen
+	sudo locale-gen
+	echo $LANG
+	cat /etc/default/locale
+	cat /etc/locale.gen
+	locale
+	cat /etc/default/locale
+	localectl status
+	sudo update-locale LANG=$_lang LANGUAGE
+	localectl status
+	echo 'Please re-login to update locale'
+fi
diff --git a/config_space/sid/files/root/provision_nftables.sh/AZURE b/config_space/sid/files/root/provision_nftables.sh/AZURE
new file mode 100644
index 0000000..3c7d3ed
--- /dev/null
+++ b/config_space/sid/files/root/provision_nftables.sh/AZURE
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+sudo apt install -y nftables
+
+IP=`echo $SSH_CLIENT | cut -d ' ' -f 1`
+cat <<EOF | sudo tee /etc/nftables.conf
+#!/usr/sbin/nft -f
+
+flush ruleset
+
+add table ip filter
+add chain ip filter INPUT { type filter hook input priority 0; policy drop; }
+add chain ip filter FORWARD { type filter hook forward priority 0; policy drop; }
+add chain ip filter OUTPUT { type filter hook output priority 0; policy accept; }
+add rule ip filter INPUT iifname "lo" counter accept
+add rule ip filter INPUT ct state related,established counter accept
+add rule ip filter INPUT ip saddr $IP tcp dport 22 counter accept
+EOF
+
+sudo systemctl enable nftables
+sudo systemctl start nftables
+systemctl status nftables
+
+sudo nft list ruleset
diff --git a/config_space/sid/files/root/provision_trim.sh/AZURE b/config_space/sid/files/root/provision_trim.sh/AZURE
new file mode 100644
index 0000000..a95aeca
--- /dev/null
+++ b/config_space/sid/files/root/provision_trim.sh/AZURE
@@ -0,0 +1,36 @@
+#!/bin/bash
+
+# ssh
+sudo rm /root/.ssh
+
+cp /etc/ssh/sshd_config /tmp/
+sudo sed -i 's/X11Forwarding yes/X11Forwarding no/' /etc/ssh/sshd_config
+
+grep AddressFamily /etc/ssh/sshd_config
+grep PermitRootLogin /etc/ssh/sshd_config
+cat <<EOF | sudo tee -a /etc/ssh/sshd_config
+
+AddressFamily inet
+PermitRootLogin no
+EOF
+
+diff /etc/ssh/sshd_config /tmp/sshd_config
+
+sudo ss -ntap | grep ssh
+sudo systemctl reload ssh
+sudo ss -ntap | grep ssh
+
+# services
+systemctl list-units --type=service --state=running
+
+for i in walinuxagent hv-kvp-daemon getty@tty1 serial-getty@ttyS0; do
+    sudo systemctl stop $i
+    sudo systemctl disable $i
+done
+
+systemctl list-units --type=service --state=running
+
+# packages
+sudo apt purge -y waagent iptables rsyslog hyperv-daemons
+sudo apt autoremove -y
+sudo rm -rf /var/lib/waagent /etc/waagent.conf /var/log/waagent.log
diff --git a/config_space/sid/hooks/partition.GRUB_CLOUD_AMD64 b/config_space/sid/hooks/partition.GRUB_CLOUD_AMD64
index 2ed97e9..3d2ea80 100755
--- a/config_space/sid/hooks/partition.GRUB_CLOUD_AMD64
+++ b/config_space/sid/hooks/partition.GRUB_CLOUD_AMD64
@@ -5,7 +5,7 @@ touch $LOGDIR/skip.partition
 set -- $disklist
 device=/dev/$1
 device_root="${device}p1"
-device_efi="${device}p15"
+device_efi="${device}p9"
 
 wait_for_device() {
   for s in $(seq 10); do
@@ -21,9 +21,9 @@ label: gpt
 unit: sectors
 
 # BIOS boot
-p14 : start=2048, size=6144, type=21686148-6449-6E6F-744E-656564454649
+p8 : start=2048, size=6144, type=21686148-6449-6E6F-744E-656564454649
 # EFI system
-p15 : start=8192, size=253952, type=C12A7328-F81F-11D2-BA4B-00A0C93EC93B
+p9 : start=8192, size=253952, type=C12A7328-F81F-11D2-BA4B-00A0C93EC93B
 # Linux
 p1 : start=262144, type=0FC63DAF-8483-4772-8E79-3D69D8477DE4
 EOF
diff --git a/src/debian_cloud_images/resources/image.yaml b/src/debian_cloud_images/resources/image.yaml
index 36dbe59..1d499ff 100644
--- a/src/debian_cloud_images/resources/image.yaml
+++ b/src/debian_cloud_images/resources/image.yaml
@@ -45,7 +45,7 @@ releases:
   basename: bookworm
   id: '12'
   baseid: '12'
-  fai_classes: [BOOKWORM, EXTRAS]
+  fai_classes: [BOOKWORM]
   arch_supports_linux_image_cloud: [amd64, arm64]
 - name: bookworm-backports
   basename: bookworm-backports
@@ -62,8 +62,8 @@ releases:
 
 vendors:
 - name: azure
-  fai_classes: [AZURE, IPV6_DHCP]
-  size: 30
+  fai_classes: [AZURE, DISABLE_IPV6]
+  size: 2
   use_linux_image_cloud: true
 - name: ec2
   fai_classes: [EC2, IPV6_DHCP]
